### PR TITLE
Add a script to build known Fuchsia artfacts into a bucket suitable for upload.

### PR DESCRIPTION
**Pull request by chinmaygarde**

### CODE REVIEW

**Comment by chinmaygarde** [comment]
WIP. Just creates the bucket for now. Will wire it up to CIPD for upload.

File: build/fuchsia/build_fuchsia_artifacts.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+"""
+
+import argparse
+import subprocess
+import os
+import errno
+import sys
+import shutil
+
+src_root_dir = os.path.abspath(os.path.join(os.path.realpath(__file__), '..', '..', '..'))
```
Comment (gaaclarke) [inline build/fuchsia/build_fuchsia_artifacts.py]: Global variables should have a "_" prepended: http://google.github.io/styleguide/pyguide.html#25-global-variables

File: build/fuchsia/build_fuchsia_artifacts.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+  assert os.path.exists(os.path.join(out_dir, variant_dir))
+
+  RunExecutable([
+    'autoninja',
+    '-C',
+    os.path.join(out_dir, variant_dir)
+  ] + targets)
+
+
+def RemoveDirectory(path):
```
Comment (gaaclarke) [inline build/fuchsia/build_fuchsia_artifacts.py]: RemoveDirectoryIfExists?

File: build/fuchsia/build_fuchsia_artifacts.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+  args = parser.parse_args()
+
+  common_flags = [
+    '--fuchsia',
+    # The source does not require LTO and LTO is not wired up for targets.
+    '--no-lto',
+  ]
+
+  RunGN('fuchsia_debug', common_flags + [
```
Comment (gaaclarke) [inline build/fuchsia/build_fuchsia_artifacts.py]: These variants are duplicated here at RunGN and BuildNinjaTargets.  You could make a datastructure to loop over to remove this duplication.

File: build/fuchsia/build_fuchsia_artifacts.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+  CopyToBucket('fuchsia_debug/icudtl.dat', 'flutter/debug/icudtl.dat')
+  CopyToBucket('fuchsia_profile/icudtl.dat', 'flutter/profile/icudtl.dat')
+  CopyToBucket('fuchsia_release/icudtl.dat', 'flutter/release/icudtl.dat')
+
+
+def main():
+  parser = argparse.ArgumentParser()
+
+  args = parser.parse_args()
```
Comment (gaaclarke) [inline build/fuchsia/build_fuchsia_artifacts.py]: unused

File: build/fuchsia/build_fuchsia_artifacts.py [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+  BuildNinjaTargets('fuchsia_debug', targets_to_build)
+  BuildNinjaTargets('fuchsia_profile', targets_to_build)
+  BuildNinjaTargets('fuchsia_release', targets_to_build)
+
+  BuildBucket()
+
+
+if __name__ == '__main__':
+  sys.exit(main())
```
Comment (gaaclarke) [inline build/fuchsia/build_fuchsia_artifacts.py]: main doesn't actually return anything.

**Pull request merged**